core/src/test/scala/unit/kafka/coordinator/GroupCoordinatorResponseTest.scala

```diff
@@ -87,7 +88,8 @@ class GroupCoordinatorResponseTest extends JUnitSuite {
     EasyMock.expect(zkUtils.getPartitionAssignmentForTopics(Seq(TopicConstants.GROUP_METADATA_TOPIC_NAME))).andReturn(ret)
     EasyMock.replay(zkUtils)
 
-    groupCoordinator = GroupCoordinator(KafkaConfig.fromProps(props), zkUtils, replicaManager, new SystemTime)
+    time = new SystemTime
```

Comment:
See the comment above. We would need a purgatory implementation which works with `MockTime`. I looked into this briefly and found that the main dependence on system time comes from the usage of `DelayQueue` in `Timer`. It appears the easiest solution is to abstract an interface for `Timer` and create a new implementation which depends on `MockTime`. We would also need to fix the definition of `expirationMs` in `DelayedOperation`, which will require a bigger refactoring so that we have a `Time` instance available wherever delayed operations are constructed. It seems doable, but it will take some work. We could also leave off the test cases for now (since the patch is trivial otherwise) and do this refactoring in a separate JIRA. Thoughts?